From 13aac521696b071d6ee3c0efefeca0e453b4f257 Mon Sep 17 00:00:00 2001
From: "Tao, Jing" <jing.tao@intel.com>
Date: Wed, 11 Mar 2015 15:16:02 +0800
Subject: [PATCH 358/360] atomisp2: clear resolution setting for node

In some situation of SDV mode, Hal would only configure
video pipe which leaves the gargabe configuration in video
nodes for capture pipe.

Clean the resolution setting for node when close.

Change-Id: Ie7e1b5a606eaf6588974155ce82e18fe5fc9248f
Tracked-On: #H1204625895
Signed-off-by: Tao, Jing <jing.tao@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/45797
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c       | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
index 2bab06d..525d67f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
@@ -895,6 +895,8 @@ static int atomisp_release(struct file *file)
 	bool acc_node;
 	struct v4l2_requestbuffers req;
 	struct v4l2_subdev_fh fh;
+	uint16_t source_pad = atomisp_subdev_source_pad(vdev);
+	struct v4l2_rect clear_compose = {0};
 	int ret = 0;
 
 	v4l2_fh_init(&fh.vfh, vdev);
@@ -1011,6 +1013,11 @@ subdev_uninit:
 		dev_err(isp->dev, "Failed to power off device\n");
 
 done:
+	atomisp_subdev_set_selection(&asd->subdev, &fh,
+				V4L2_SUBDEV_FORMAT_ACTIVE,
+				source_pad,
+				V4L2_SEL_TGT_COMPOSE, 0,
+				&clear_compose);
 	rt_mutex_unlock(&isp->mutex);
 	mutex_unlock(&isp->streamoff_mutex);
 
-- 
1.9.1

